apps/settings/js/panels/call_barring/panel.js

```diff
@@ -0,0 +1,210 @@
+     *   message: [string]      // optional, new message for the description
+     * }
+     */
+    function _updateCallBarringItem(item, newStatus) {
+      var descText = item.querySelector('small');
+      var input = item.querySelector('input');
+
+      // disable the item
+      if (typeof newStatus.disabled === 'boolean') {
+        newStatus.disabled ?
+          item.setAttribute('aria-disabled', true) :
+          item.removeAttribute('aria-disabled');
+
+        if (input) {
+          input.disabled = newStatus.disabled;
+        }
+      }
+
+      // update the input value
+      if (input && typeof newStatus.checked === 'boolean') {
+        input.checked = newStatus.checked;
+      }
+
+      // update the description
+      function inputValue() {
+        return input && input.checked ? 'enabled' : 'disabled';
+      }
+      if (descText) {
+        var text = _updating ? 'callSettingsQuery' : inputValue();
+        navigator.mozL10n.setAttributes(descText, text);
+      }
+    }
+
+    /**
+     * Shows the passcode input screen for the user to introduce the PIN
+     * needed to activate/deactivate a service
+     */
+    function _callBarringClick(evt) {
+      var input = evt.target;
+
+      // do not change the UI, let it be managed by the data model
+      evt.preventDefault();
+      // Show passcode screen
+      _passcodeScreen.show().then(function confirmed(passcode) {
+        // passcode screen confirmed
+        var inputID = input.parentNode.parentNode.id;
+
+        var setting = inputID.substring(6);
+        _updateMobileConnection();
+        _callBarring.set(_mobileConnection, setting, passcode).catch(
+          function error(err) {
+          // err = { name, message }
+          var toast = {
+            messageL10nId: 'callBarring-update-item-error',
+            messageL10nArgs: {'error': err.name || 'unknown'},
+            latency: 2000,
+            useTransition: true
+          };
+          Toaster.showToast(toast);
+        });
+      }).catch(function canceled() {
+        // passcode screen canceled, nothing to do yet
+      });
+    }
+
+    return SettingsPanel({
+      onInit: function cb_onInit(panel) {
+        _cbSettings = {
+          baoc: document.getElementById('li-cb-baoc'),
+          boic: document.getElementById('li-cb-boic'),
+          boicExhc: document.getElementById('li-cb-boicExhc'),
+          baic: document.getElementById('li-cb-baic'),
+          baicR: document.getElementById('li-cb-baicR')
+        };
+
+        for (var i in _cbSettings) {
+          _cbSettings[i].querySelector('input').
+            addEventListener('click', _callBarringClick);
+        }
+
+        _updateMobileConnection();
+        _passcodeScreen.init();
+      },
+
+      onBeforeShow: function cb_onBeforeShow() {
+        _refresh = true;
+        _updating = false;
+
+        window.addEventListener('panelready', refresh_on_load);
+
```

Comment:
Although it displays the correct settings in the end but users would see the settings change from the previous sim card to the current one (and which takes several seconds). For example, assume all of the call barring options are enabled for sim 1. When users switch to the panel for sim 2 (where all the cal barring options are disabled) , they would see the check boxes change from checked to unchecked one by one, which is not desired. I think setting the check boxes to unchecked by default is the simplest way for now.